src/classes/UTIL_Describe.cls

```diff
@@ -374,83 +374,84 @@ public class UTIL_Describe {
     /*******************************************************************************************************
     * @description Cache for the record type DeveloperName-to-ID map.
     */
-    private static Map<String, Id> mapRecordTypes;
-    
+    private static Map<String, Map<String, Id>> mapRecordTypes = new Map<String, Map<String, Id>>();
```

Comment:
If you are changing the type of this variable, you'll have to update its description.